cmd/eval-dev-quality/cmd/report.go

```diff
@@ -0,0 +1,109 @@
+package cmd
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	pkgerrors "github.com/pkg/errors"
+
+	"github.com/symflower/eval-dev-quality/evaluate/report"
+	"github.com/symflower/eval-dev-quality/log"
+)
+
+// Report holds the "report" command.
+type Report struct {
+	// EvaluationPaths holds the list of file paths where the results of previous evaluations are stored.
+	EvaluationPaths []string `long:"evaluation-path" description:"File path for an evaluation CSV file where the results of a previous evaluation are stored."`
+	// ResultPath holds the directory path where the overall results should be written to.
+	ResultPath string `long:"result-path" description:"Directory path were the overall results are written to."`
+
+	// logger holds the logger of the command.
+	logger *log.Logger
+}
+
+var _ SetLogger = (*Evaluate)(nil)
+
+// SetLogger sets the logger of the command.
+func (command *Report) SetLogger(logger *log.Logger) {
+	command.logger = logger
+}
+
+// Execute executes the command.
+func (command *Report) Execute(args []string) (err error) {
+	evaluationCSVFilePath, err := os.OpenFile(filepath.Join(command.ResultPath, "evaluation.csv"), os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0755)
+	if err != nil {
+		command.logger.Panicf("ERROR: %s", err)
+	}
+	defer evaluationCSVFilePath.Close()
+
+	evaluationCSVWriter := bufio.NewWriter(evaluationCSVFilePath)
+	if _, err = evaluationCSVWriter.WriteString(strings.Join(report.EvaluationHeader(), ",") + "\n"); err != nil {
+		command.logger.Panicf("ERROR: %s", err)
+	}
+
+	var allRecords []string
+	for _, evaluationPath := range command.EvaluationPaths {
+		records, err := recordsFromGlobPattern(evaluationPath)
+		if err != nil {
+			command.logger.Panicf("ERROR: %s", err)
+		}
+		allRecords = append(allRecords, records...)
+	}
+
+	sort.Strings(allRecords)
+	for _, record := range allRecords {
+		if _, err = evaluationCSVWriter.WriteString(record + "\n"); err != nil {
+			command.logger.Panicf("ERROR: %s", err)
+		}
+	}
+	evaluationCSVWriter.Flush()
+
+	return nil
+}
+
+// recordsFromGlobPattern takes a glob pattern and return the list of records from the evaluation CSV files found with the pattern.
+func recordsFromGlobPattern(evaluationGlobPattern string) (records []string, err error) {
+	evaluationCSVFilePaths, err := filepath.Glob(evaluationGlobPattern)
+	if err != nil {
+		return nil, pkgerrors.WithStack(err)
+	} else if len(evaluationCSVFilePaths) == 0 {
+		return nil, pkgerrors.Errorf("no files matched the pattern %q", evaluationGlobPattern)
+	}
+
+	for _, evaluationCSVFilePath := range evaluationCSVFilePaths {
+		if filepath.Base(evaluationCSVFilePath) != "evaluation.csv" {
```

Comment:
Move the check for the GlobPattern to the top and do an early exit. With a message like `the path needs to end with `evaluation.csv`